src/item_proxy.rs

```diff
@@ -0,0 +1,62 @@
+//Copyright 2020 secret-service-rs Developers
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! A dbus proxy for speaking with secret service's `Item` Interface.
+
+use serde::{Deserialize, Serialize};
+use zbus;
+use zbus_macros::dbus_proxy;
+use zvariant::{Dict, ObjectPath, OwnedObjectPath};
+use zvariant_derive::Type;
+
+/// This will derive ItemInterfaceProxy
+#[dbus_proxy(
+    interface = "org.freedesktop.Secret.Item",
+)]
+trait ItemInterface {
+    fn delete(&self) -> zbus::Result<OwnedObjectPath>;
+
+    /// returns `Secret`
+    fn get_secret(&self, session: ObjectPath) -> zbus::Result<SecretStruct>;
+
+    fn set_secret(&self, secret: SecretStructInput) -> zbus::Result<()>;
+
+    #[dbus_proxy(property)]
+    fn locked(&self) -> zbus::fdo::Result<bool>;
+
+    // Looks like the Dict has to be transformed into HashMap<String, String> in separate step?
+    #[dbus_proxy(property)]
+    fn attributes(&self) -> zbus::fdo::Result<Dict>;
```

Comment:
If the attributes are known, I would make it a struct that derives SerializeDict/DesrializeDict from zvariant_derive, see https://gitlab.freedesktop.org/zeenix/zbus/-/merge_requests/145

Reply:
I believe the attributes are arbitrary, set by the user in this case.

Reply:
Ah yes, my bad

Reply:
`Dict` should ideally not be needed except when dealing with Variants (`zvariant::Value`). `HashMap<String, String>` doesn't work here? It should.

Reply:
Ok will change. I think I had tried this earlier, but may have been distracted by an unrelated proc macro error.

Reply:
I'm getting this error when I try to replace `Dict` with `HashMap`:
```
error[E0277]: the trait bound `HashMap<std::string::String, std::string::String>: From<OwnedValue>` is not satisfied
  --> src/proxy/item.rs:22:1
   |
22 | / #[dbus_proxy(
23 | |     interface = "org.freedesktop.Secret.Item",
24 | | )]
   | |__^ the trait `From<OwnedValue>` is not implemented for `HashMap<std::string::String, std::string::String>`
   |
   = note: required because of the requirements on the impl of `Into<HashMap<std::string::String, std::string::String>>` for `OwnedValue`
   = note: required because of the requirements on the impl of `TryFrom<OwnedValue>` for `HashMap<std::string::String, std::string::String>`
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error; 2 warnings emitted

For more information about this error, try `rustc --explain E0277`.
```

Reply:
That seems like a bug in zbus/zvariant (we should provide the needed conversion here). I'll look into it. BTW, do ensure you are actualling getting `HashMap<String, String>` here as usually you get `HashMap<String, OwnedValue>` or `HashMap<OwnedObjectPath, OwnedValue>` in such cases.

Reply:
Ah, actually the thing is that you need to wrap it in a [`Result`](https://gitlab.freedesktop.org/zeenix/zbus/-/merge_requests/195/diffs#6bbc4c02ed8264a440efa6ba83673c170d323eb3_660_660).

Reply:
oh, i missed the fact that this is a property, not a normal method. I'll check again. :)

Reply:
OK, so there were missing conversions but I've[ added them now](https://gitlab.freedesktop.org/zeenix/zbus/-/merge_requests/196). Will roll out a zvariant release soon with these fixes.

Reply:
Just released zvariant 2.4.0 that solves your problem. :)